keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/MatrixDeterminantVertex.java

```diff
@@ -42,7 +42,7 @@ public PartialDerivatives forwardModeAutoDifferentiation(Map<Vertex, PartialDeri
 
         PartialDerivatives derivativeOfOutputsWithRespectToInputs = derivativeOfOutputsWithRespectToSelf
             .multiplyBy(inputVertex.getValue().determinant())
-            .multiplyAlongWrtDimensions(inverseTranspose, this.inputVertex.getShape());
+            .multiplyAlongWrtDimensions(inverseTranspose, this.getShape());
```

Comment:
It was wrong before. It never causes a problem because before, the shape of the input was the same as the output (N,M) -> (1, 1) but is now (N,M) -> ().